ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/HiveWrapper.java

```diff
@@ -54,8 +55,11 @@ public HiveWrapper(Hive db, String dbName, long lastReplId) {
     return new Tuple<>(functionForSpec, () -> db.getDatabase(dbName));
   }
 
-  public Tuple<Table> table(final String tableName) throws HiveException {
-    return new Tuple<>(functionForSpec, () -> db.getTable(dbName, tableName));
+  public Tuple<Table> table(final String tableName, HiveConf conf) throws HiveException {
+    // Column statistics won't be accurate if we are dumping only metadata
+    boolean getColStats = !conf.getBoolVar(HiveConf.ConfVars.REPL_DUMP_METADATA_ONLY);
```

Comment:
If it is metadata only dump, then need to reset all the table parameters related to stats to false.

Reply:
Added a testcase for the same. It's passing except in one peculiar case when there is an ALTER TABLE which changes the column names. TestReplicationScenariosAcrossInstances.testIncrementalMetadataReplication() changes the name of column “I” in table3 to “a”. On primary the column name seen in getTable() is still “I” but on replica it is “a”. Whereas that in COLUMN_STATS_ACCURATE is “I” on replica as well as primary. Looks like a bug in existing code. Need more investigation.

Reply:
yep... looks to be bug in base code.

Reply:
For now I have commented out the test. Once I have a concrete understanding of the bug, I will create an BUG in JIRA.